sw/device/lib/dif/dif_clkmgr.h

```diff
@@ -0,0 +1,200 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CLKMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CLKMGR_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/clkmgr/doc/">Clock manager</a> Device Interface
+ * Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_clkmgr_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifClkmgrToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifClkmgrToggleDisabled,
+} dif_clkmgr_toggle_t;
+
+/**
+ * Hardware instantiation parameters for clock manager.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_clkmgr_params {
+  /**
+   * The base address for the clock manager hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_clkmgr_params_t;
+
+/**
+ * A handle to clock manager.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_clkmgr {
+  dif_clkmgr_params_t params;
+} dif_clkmgr_t;
+
+/**
+ * The result of a clock manager operation.
+ */
+typedef enum dif_clkmgr_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifClkmgrOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifClkmgrError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifClkmgrBadArg = 2,
+} dif_clkmgr_result_t;
+
+/**
+ * An Index of a "Gateable" Clock.
+ *
+ * "Gateable" clocks are under full software control: they can be enabled and
+ * disabled by software, which directly starts and stops the identified clock.
+ */
+typedef uint32_t dif_clkmgr_gateable_clock_t;
+
+/**
+ * An Index of a "Hintable" Clock.
+ *
+ * "Hintable" clocks are under partial software control: software can suggest
+ * they are gated, but the clock manager may delay gating the peripheral, or may
+ * ignore the request altogether.
+ */
```

Comment:
Is this completely correct?

The documentation says the following:
```
        Clock hint for software gateable clocks.
        These clocks are not fully controlled by software.

        For disable, software only provides a hint, and hardware determines the final clock state based on the
        hint and whether the block in question is idle.
```

Which only describes a special restriction in regard to the `disable` operation, says nothing about `enable` operation.

The individual bits description also says the following:
```
      fields: [
        {
          bits: "0",
          name: "CLK_MAIN_AES_HINT",
          resval: 1,
          desc: '''
            0 CLK_MAIN_AES can be disabled.
            1 CLK_MAIN_AES is enabled.
          '''
        }
```
Which  only for the `disable` has `can` in the description. The `enable` description is the same as for the gateable clocks.